community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -424,7 +424,8 @@ public GBPTree( PageCache pageCache, File indexFile, Layout<KEY,VALUE> layout, i
             this.pagedFile = openOrCreate( pageCache, indexFile, tentativePageSize, layout );
             this.pageSize = pagedFile.pageSize();
             closed = false;
-            this.bTreeNode = new TreeNodeFixedSize<>( pageSize, layout );
+            this.bTreeNode = layout.fixedSize() ? new TreeNodeFixedSize<>( pageSize, layout )
```

Comment:
I was thinking about selection of `TreeNode` and the version header. Nowhere will the version somehow contain which impl of `TreeNode` that the tree is using, will it? The selection will be on `Layout#fixedSize()` which theoretically can change between neo4j versions (perhaps not practically though). Is this something we should bring into the version header as well? Maybe mentioning selection of `TreeNode` in the version header will have future benefits too. What are your thoughts?

Reply:
Yes, that part was forgotten / ignored for now. But we should definitely make TreeNode version part of complete version in some way!